onnxruntime/core/optimizer/double_qdq_pairs_remover.cc

```diff
@@ -45,21 +46,16 @@ bool DoubleQDQPairsRemover::IsNodeRemovable(
       self->OpType() != "DequantizeLinear" ||
       self->GetInputEdgesCount() != 1 ||
       self->GetOutputEdgesCount() != 1 ||
-      self->InputDefs().size() != InputIndex::TOTAL_COUNT ||
       graph.NodeProducesGraphOutput(*self)) {
     return false;
   }
 
-  // Type is either "tensor(uint8)" or  "tensor(int8)"
-  const auto self_zp_type = *self->InputDefs()[InputIndex::ZERO_POINT_ID]->Type();
-  // child should be a Q, and have only one child, have the same type as self, and cannot be a graph output
+  // child should be a Q, and have only one child, and cannot be a graph output
   child_index = self->OutputEdgesBegin()->GetNode().Index();
   const Node* child = graph.GetNode(child_index);
   if (child == nullptr ||
       child->OpType() != "QuantizeLinear" ||
       child->GetOutputEdgesCount() != 1 ||
-      child->InputDefs().size() != InputIndex::TOTAL_COUNT ||
-      *child->InputDefs()[InputIndex::ZERO_POINT_ID]->Type() != self_zp_type ||
```

Comment:
Removed these conditions because the call IsQDQPairSupported down below will perform these checks. They don't need to happen now.